init.go

```diff
@@ -0,0 +1,245 @@
+			if c.verbose {
+				fmt.Println(dep.ImportPath)
+			}
+			c.deps[dep.ImportPath] = dep
+
+		} else {
+			// we put ignored packages in the dep map as nils just so we'll scan them once for their deps
+			c.deps[dep.ImportPath] = nil
+		}
+
+		// we find deps even on "ignored" packages - we want their deps
+		c.findDeps(dep)
+	}
+
+	return nil
+}
+
+// findDepsInDir finds all the deps for the packages under the dir in path.
+// this is used to recursively traverse all packages in a project
+func (c *depFinder) findDepsInDir(pth string) error {
+
+	c.wd = pth
+	os.Chdir(c.wd)
+
+	pkgs, err := c.lister.List("")
+	if err != nil {
+		return err
+	}
+	for _, p := range pkgs {
+		c.findDeps(p)
+	}
+
+	files, err := ioutil.ReadDir(pth)
+
+	if err != nil {
+		return err
+	}
+
+	for _, file := range files {
+		if !file.IsDir() {
+			continue
+		}
+
+		if strings.HasPrefix(file.Name(), ".") || strings.HasPrefix(file.Name(), "_") {
+			continue
+		}
+
+		err := c.findDepsInDir(path.Join(pth, file.Name()))
+		if err != nil {
+			fmt.Println("error traversing", file.Name(), ": ", err)
+			return err
+		}
+
+	}
+
+	return nil
+
+}
+
+// FindDependencies traverses the current WD and retrieves a dependency map (package => revision)
+// for all 3rd party libraries.
+// VCS refs are retrieved only if the vcsRefs flag is set to true
+func (c *depFinder) FindDependencies(vcsRefs bool) (map[string]string, error) {
+	err := c.findDepsInDir(c.wd)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := make(map[string]string)
+	if c.verbose && vcsRefs {
+		fmt.Println("Finding VCS refs for packages")
+	}
+
+	for dep, pkg := range c.deps {
+
+		// nil pkg's are placeholders for not entering the same dir twice, we ignore them
+		if pkg == nil {
+			continue
+		}
+		tag := ""
+		if vcsRefs {
+			_, vc, err := VCSForImportPath(dep)
+
+			if err == nil {
+				tag, _ = vc.Identify(pkg.Dir)
+			} else {
+				fmt.Printf("Error extracting tag for %s: %s\n", dep, err)
+			}
+
+		}
+		ret[dep] = tag
+
+	}
+
+	return ret, nil
+}
+
+func runInit(c *cli.Context) {
+
+	//the first arg can be the path of a project
+	if len(c.Args()) > 0 {
+		pth := c.Args().First()
+		if pth != "" && pth != "." {
+			err := os.Chdir(pth)
+			if err != nil {
+				fmt.Printf("Invalid path %s (%s)\n", pth, err)
+				os.Exit(1)
+			}
+		}
+	}
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	ctx := &depFinder{
+		deps:    make(map[string]*pkg),
+		ignored: c.StringSlice("ignored"),
+		wd:      cwd,
+		verbose: c.Bool("verbose"),
+		lister: pkgLister{
+			Env: os.Environ(),
+		},
+	}
+
+	deps, err := ctx.FindDependencies(c.Bool("refs"))
+
+	// create a manifest from the gathered data
+	manifest := Manifest{
+		App: ManifestApp{
+			Name:    path.Base(cwd),
+			Authors: []string{"Your Name <you@example.com>"},
+			Version: "0.0.1",
+		},
+		Deps: ManifestDeps(deps),
+	}
+
+	os.Chdir(cwd)
+
+	err = manifest.write()
+	if err != nil {
+		fmt.Println(err)
```

Comment:
right. I'll change that. it's weird BTW that the cli package doesn't allow you to return an error from running the app. any idea why that is?